packages/components/src/select-control/index.js

```diff
@@ -27,16 +27,16 @@ export class SelectControl extends Component {
 		super( props );
 		this.state = {
 			...initialState,
-			filteredOptions: [],
+			searchOptions: [],
```

Comment:
I think the renaming here is a good call and makes more sense for what these options actually are. 👍 

On a side note, instead of storing in state this would be benefit from some level of memoization, but I think this would need some testing with async functions and is probably outside the scope of this PR.